src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/UserTimeZoneService.cs

```diff
@@ -85,7 +89,8 @@ private async ValueTask<string> GetTimeZoneIdAsync(IUser user)
         // Do we know this user doesn't have a configured value?
         if (timeZoneId == EmptyTimeZone)
         {
-            return null;
+            // The user opted to use the site's time zone.
+            return (await _siteService.GetSiteSettingsAsync())?.TimeZoneId;
```

Comment:
I contemplated `UserTimeZoneSelector` returning an empty result in these cases instead (i.e. letting the site's time zone take precedence) but it's good API design anyway not to return `null` values if not necessary. And in this case, the user actually deliberately set a value, it shouldn't return `null`.

Reply:
When TimeZone is set in general settings and not selected in user, it gives an error again. I pulled out Main and tried it.

<img width="1152" alt="Screenshot 2024-09-21 at 21 39 16" src="https://github.com/user-attachments/assets/54ce7ef4-e67a-46ac-978a-c27e2044735e">


<img width="1152" alt="Screenshot 2024-09-21 at 21 36 14" src="https://github.com/user-attachments/assets/4c9be7f3-adad-43d3-b9c7-490867c82b71">

Reply:
You need to pull this branch, not `main`: https://github.com/Lombiq/OrchardCore/tree/issue/OCORE-200

Reply:
> You need to pull this branch, not `main`: https://github.com/Lombiq/OrchardCore/tree/issue/OCORE-200

Yes, problem solved, thank you Zoltán

Reply:
Great, thanks for checking.